packages/block-library/src/query-loop/index.php

```diff
@@ -27,6 +27,10 @@ function render_block_core_query_loop( $attributes, $content, $block ) {
 			// Unset `offset` because if is set, $wp_query overrides/ignores the paged parameter and breaks pagination.
 			unset( $query['offset'] );
 			$query = wp_parse_args( $wp_query->query_vars, $query );
+
+			if ( '' === $query['post_type'] && is_singular() ) {
```

Comment:
Is it safe to assume that `post_type` is defined? Otherwise `empty( $query['post_type'] )` check would be better.

Reply:
Fair point. Changed to use `empty()` :+1: